app/assets/javascripts/i18n.js

```diff
@@ -935,6 +976,47 @@
     return extended;
   };
 
+  // Takes an array and adds the given value between each element.
+  //
+  // Examples:
+  // interleaveValue("a", []) => []
+  // interleaveValue("a", [1]) => [1]
+  // interleaveValue("a", [1, 2]) => [1, "a", 2]
+  // interleaveValue("a", [1, 2, 3]) => [1, "a", 2, "a", 3]
+  //
+  // This function is not part of the public API
+  //
+  I18n.interleaveValue = function ( value, arr ) {
+    var first = arr.slice(0, 1); // first element in an array, or empty array
+    var rest = arr.slice(1); // rest in an array
+
+    if (rest.length) {
+      return rest.reduce(function(xs, x) {
+        return xs.concat([value, x]);
+      }, first);
+    } else {
+      return first;
+    }
+  };
+
+  // Trims array, i.e. removes unwanted value from the beginning and end
+  //
+  // Example:
+  // var splitted = "My name is ${name}".split("${name}"); => ["My name is ", ""];
+  // I18n.trimArray("", splitted); => ["My name is "];
+  I18n.trimArray = function ( unwantedValue, arr ) {
+    if (arr.length === 0) {
+      return arr;
+    } else if (arr.length === 1) {
+      return (arr[0] === unwantedValue) ? arr.slice(1) : arr.slice();
+    } else {
+      var start = (arr[0] === unwantedValue) ? 1 : 0;
+      var end = (arr[arr.length - 1] === unwantedValue) ? -1 : undefined;
+
+      return arr.slice(start, end);
+    }
+  };
```

Comment:
Usually trim functions trim recursively while this only removes a single instance of the `unwantedValue` from the start and the end. Is it ok in this case that `I18n.trimArray("", "My name is ${name}${name}".split("${name}"))` yields `["My name is ", ""]`?